src/openzaak/components/autorisaties/admin_views.py

```diff
@@ -183,33 +145,66 @@ def get_initial_for_component(
             initial.append(_initial)
 
     elif component == ComponentTypes.brc:
-        besluittype_ids = set(BesluitType.objects.values_list("id", flat=True))
         relevant_ids = set(related_objs.values())
+        _initial = {
+            "externe_typen": list(_related_objs_external.values()),
+            "related_type_selection": RelatedTypeSelectionMethods.manual_select,
+            "besluittypen": relevant_ids,
+        }
+        initial.append(_initial)
+    else:
+        # The other components do not have any extra options
+        initial.append({})
+
+    return initial
 
-        _initial = {"externe_typen": _related_objs_external}
 
-        if spec:
-            _initial["related_type_selection"] = (
-                RelatedTypeSelectionMethods.all_current_and_future
-            )
-        elif besluittype_ids == relevant_ids:
-            _initial["related_type_selection"] = RelatedTypeSelectionMethods.all_current
-        else:
+def get_initial_for_component_for_catalogus_autorisaties(
+    component: str,
+    catalogus_autorisaties: list[CatalogusAutorisatie],
+) -> List[Dict[str, Any]]:
+    initial = []
+
+    match component:
+        case ComponentTypes.zrc | ComponentTypes.drc:
+            grouped_by_va = defaultdict(list)
+            for autorisatie in catalogus_autorisaties:
+                grouped_by_va[autorisatie.max_vertrouwelijkheidaanduiding].append(
+                    autorisatie
+                )
+
+            for va, _autorisaties in grouped_by_va.items():
+                _initial = {"vertrouwelijkheidaanduiding": va}
+                catalogus_autorisaties_for_va = [
+                    catalogus_autorisatie.catalogus.pk
+                    for catalogus_autorisatie in _autorisaties or []
+                ]
+                _initial.update(
+                    {
+                        "related_type_selection": RelatedTypeSelectionMethods.select_catalogus,
+                        "catalogi": catalogus_autorisaties_for_va,
+                    }
+                )
+                initial.append(_initial)
+        case ComponentTypes.brc:
+            _initial = {}
             _initial.update(
                 {
```

Comment:
```suggestion
            _initial = {
```

or even maybe `initial = [{"related_type_selection": <...>}]`, since if it's `brc` it would be always a list with 1 member inside